superset/utils/webdriver.py

```diff
@@ -176,22 +176,46 @@ def get_screenshot(
         sleep(selenium_headstart)
 
         try:
-            logger.debug("Wait for the presence of %s", element_name)
-            element = WebDriverWait(driver, self._screenshot_locate_wait).until(
-                EC.presence_of_element_located((By.CLASS_NAME, element_name))
-            )
+            try:
+                # page didn't load
+                logger.debug(
+                    "Wait for the presence of %s at url: %s", element_name, url
+                )
+                element = WebDriverWait(driver, self._screenshot_locate_wait).until(
+                    EC.presence_of_element_located((By.CLASS_NAME, element_name))
+                )
```

Comment:
These are most likely going to be network errors, i.e., server is down or url is incorrect, or perhaps permissions - user can't access the page.